modular_skyrat/modules/modular_items/lewd_items/code/lewd_helpers/human.dm

```diff
@@ -210,106 +187,80 @@
+	if(!genital)
+		return FALSE
+	switch(required_state)
+		if(REQUIRE_GENITAL_ANY)
+			return TRUE
+		if(REQUIRE_GENITAL_EXPOSED)
+			return !get_item_by_slot(ITEM_SLOT_EARS)
+		if(REQUIRE_GENITAL_UNEXPOSED)
+			return get_item_by_slot(ITEM_SLOT_EARS)
+		else
+			return TRUE
 
+/// Returns true if the human has accessible eyes for the parameter. Accepts any of the `REQUIRE_GENITAL_` defines.
 /mob/living/carbon/human/proc/has_eyes(required_state = REQUIRE_GENITAL_ANY)
-	var/obj/item/organ/peepee = getorganslot(ORGAN_SLOT_EYES)
-	if(peepee)
-		switch(required_state)
-			if(REQUIRE_GENITAL_ANY)
-				return TRUE
-			if(REQUIRE_GENITAL_EXPOSED)
-				if(get_item_by_slot(ITEM_SLOT_EYES))
-					return FALSE
-				else
-					return TRUE
-			if(REQUIRE_GENITAL_UNEXPOSED)
-				if(!get_item_by_slot(ITEM_SLOT_EYES))
-					return FALSE
-				else
-					return TRUE
-			else
-				return TRUE
-	return FALSE
+	var/obj/item/organ/genital = getorganslot(ORGAN_SLOT_EYES)
+	if(!genital)
+		return FALSE
+	switch(required_state)
+		if(REQUIRE_GENITAL_ANY)
+			return TRUE
+		if(REQUIRE_GENITAL_EXPOSED)
+			return !get_item_by_slot(ITEM_SLOT_EYES)
+		if(REQUIRE_GENITAL_UNEXPOSED)
+			return get_item_by_slot(ITEM_SLOT_EYES)
+		else
+			return TRUE
 
+/// Returns true if the human has accessible eye sockets for the parameter. Accepts any of the `REQUIRE_GENITAL_` defines.
 /mob/living/carbon/human/proc/has_eyesockets(required_state = REQUIRE_GENITAL_ANY)
-	var/obj/item/organ/peepee = getorganslot(ORGAN_SLOT_EYES)
-	if(!peepee)
-		switch(required_state)
-			if(REQUIRE_GENITAL_ANY)
-				return TRUE
-			if(REQUIRE_GENITAL_EXPOSED)
-				if(get_item_by_slot(ITEM_SLOT_EYES))
-					return FALSE
-				else
-					return TRUE
-			if(REQUIRE_GENITAL_UNEXPOSED)
-				if(!get_item_by_slot(ITEM_SLOT_EYES))
-					return FALSE
-				else
-					return TRUE
-			else
-				return TRUE
-	return FALSE
+	var/obj/item/organ/genital = getorganslot(ORGAN_SLOT_EYES)
+	if(!genital)
+		return FALSE
+	switch(required_state)
+		if(REQUIRE_GENITAL_ANY)
+			return TRUE
+		if(REQUIRE_GENITAL_EXPOSED)
+			return !get_item_by_slot(ITEM_SLOT_EYES)
+		if(REQUIRE_GENITAL_UNEXPOSED)
+			get_item_by_slot(ITEM_SLOT_EYES)
+		else
+			return TRUE
 
 
 /*
 *	This code needed for changing character's gender by chems
 */
 
+/// Sets the gender of the human, respecting prefs unless it's forced. Do not force in non-admin operations.
 /mob/living/carbon/human/proc/set_gender(ngender = NEUTER, silent = FALSE, update_icon = TRUE, forced = FALSE)
-	if(forced || (!ckey || client?.prefs.read_preference(/datum/preference/toggle/erp/gender_change)))
-		gender = ngender
-		return TRUE
-	return FALSE
-
-/mob/living/carbon/human/set_gender(ngender = NEUTER, silent = FALSE, update_icon = TRUE, forced = FALSE)
 	var/bender = !(gender == ngender)
-	. = ..()
-	if(!.)
-		return
+	if(!forced && !client?.prefs?.read_preference(/datum/preference/toggle/erp/gender_change))
+		return FALSE
```

Comment:
`!forced` is going to be TRUE basically all the time, leave it as the last thing checked here, for the sake of efficiency.
```suggestion
	if(!client?.prefs?.read_preference(/datum/preference/toggle/erp/gender_change) && !forced)
		return FALSE

```